common/src/main/java/com/viaversion/viaversion/data/entity/EntityTrackerBase.java

```diff
@@ -102,15 +102,15 @@ public void clearEntities() {
     }
 
     @Override
-    public int clientEntityId() {
+    public Integer clientEntityId() {
```

Comment:
I'd prefer still returning a primitive here and instead adding a hasClientEntityId method

Reply:
Because of backwards compatibility? Then I still have to return -1 and it generally is more likely to be used wrong as people would always need to remember wrapping it around hasClientEntityId.

Reply:
It could throw an exception if being called when the client entity id not set to avoid using the method wrongly

Reply:
Good point, removed

Reply:
Most packets will also throw on the client if join game hasn't been sent yet, or it'll be Via failing to track it. So hasClientEntityId won't always be necessary, but throwing is necessary